plotly/offline.py

```diff
@@ -0,0 +1,178 @@
+""" Plotly Offline
+    A module to use Plotly's graphing library with Python
+    without connecting to a public or private plotly enterprise
+    server.
+"""
+from __future__ import absolute_import
+
+import uuid
+import json
+import os
+import requests
+
+import plotly.plotly as py
+from plotly import utils
+from plotly.exceptions import PlotlyError
+
+PLOTLY_OFFLINE_DIRECTORY = plotlyjs_path = os.path.expanduser(
+    os.path.join(*'~/.plotly/plotlyjs'.split('/')))
+PLOTLY_OFFLINE_BUNDLE = os.path.join(PLOTLY_OFFLINE_DIRECTORY,
+                                     'plotly-ipython-offline-bundle.js')
+
+__PLOTLY_OFFLINE_INITIALIZED = False
+
+
+def download_plotlyjs(download_url):
+    if not os.path.exists(plotlyjs_path):
+        os.makedirs(plotlyjs_path)
```

Comment:
can you check for write permissions and gracefully exit here? I think the tools.py module defines it:
https://github.com/plotly/python-api/blob/master/plotly/tools.py#L61-76

`_file_permissions`